### PR TITLE
Add PydanticAI sandbox command toolset

### LINKED ISSUE #418: Wire PydanticAI code execution through active stack sandbox
## Parent

- #415

## What to build

Wire PydanticAI code execution through Kitaru's active-stack sandbox helper.

PydanticAI is a good first adapter because Kitaru already wraps model calls, tool calls, MCP calls, and HITL tools. This issue should add an explicit PydanticAI-friendly sandbox tool or toolset that executes commands through the active ZenML sandbox instead of the host process.

## Acceptance criteria

- [ ] Audit the existing PydanticAI tool/toolset wrapping paths and decide the cleanest place for a sandbox-backed tool/toolset.
- [ ] Add a Kitaru-provided PydanticAI sandbox tool or toolset.
- [ ] Execute commands through the active Kitaru/ZenML sandbox component.
- [ ] Keep command calls visible in Kitaru's existing tool-call tracking/checkpointing behavior.
- [ ] Add tests for successful command execution, non-zero exit, missing sandbox component, and replay/checkpoint behavior.
- [ ] Add or update a runnable PydanticAI example.
- [ ] Update PydanticAI adapter docs with the supported usage and safety limitations.

## Blocked by

- #415
- #416
- #417

### PR DESCRIPTION
**Pull request by strickvl**
Fixes #418

## What changed

This PR adds a PydanticAI-facing sandbox command toolset on top of the sandbox stack/runtime work from `feature/sandbox-stack-component-base`.

- adds `sandbox_command_toolset(...)` under `kitaru.adapters.pydantic_ai`
- exposes one model-facing tool, `run_sandbox_command(command, cwd=None)`
- routes command execution through `kitaru.run_sandbox_command(...)`, so the active stack sandbox executes the command instead of the host process
- returns a narrow `SandboxCommandToolResult` with stdout, stderr, exit code, truncation flags, and cleanup status
- keeps `env` out of the model-facing tool arguments so environment values are not accidentally captured as tool-call args
- validates `max_chars` and `cleanup` when the toolset is constructed, before the model starts running
- keeps non-zero command exits as completed tool results rather than adapter failures
- documents the feature and adds a runnable PydanticAI example
- registers the new example in `examples/example-coverage.yaml`
- adds provider-backed release smoke coverage in `scripts/smoke-test.sh`, gated on both `OPENAI_API_KEY` and an active stack with exactly one sandbox component
- recommends disabling cache for the sandbox command tool checkpoint because shell commands can have side effects

## Reviewer Notes

The important path is deliberately small. A user attaches `sandbox_command_toolset(...)` to a PydanticAI agent; `KitaruAgent` then sees it as an ordinary PydanticAI `FunctionToolset`. When the model calls `run_sandbox_command`, the existing `KitaruFunctionToolset` / `KitaruToolset.call_tool(...)` path records and checkpoints the tool call, and the tool body calls the shared sandbox helper from #417.

The main thing to inspect is that this does **not** introduce a second sandbox/session implementation inside the PydanticAI adapter. One tool call means one call to `kitaru.run_sandbox_command(...)`; that helper still does active-stack lookup, session creation, command execution, output collection, and cleanup.

Also check the safety story. The model can choose `command` and optional `cwd`, but cannot provide `env`. The factory owner controls `max_chars` and cleanup policy (`destroy` by default, or `close` if the caller wants to leave the sandbox session open according to the runtime helper semantics). The adapter default is 20,000 chars so a noisy command does not send a 1 MiB tool result back through the model by default.

The docs now spell out the risk in concrete terms: the local sandbox is a local subprocess, not a security boundary. If a model or untrusted prompt controls the command, it can ask the sandbox process to print visible files, environment variables, network-reachable data, or credentials to stdout/stderr, and that output can be returned to the model. For untrusted prompts or models, use an isolated sandbox provider with minimal credentials.

For replay, the docs and example show the recommended default:

```python
tool_checkpoint_config_by_name={
    SANDBOX_COMMAND_TOOL_NAME: {"cache": False},
}
```

That means a replay runs the command again instead of silently reusing an old command result. This matters for commands like `pip install ...`, file writes, database migrations, or anything else where the command changes the sandbox state.

The example intentionally keeps per-tool checkpoints visible. It runs the agent directly in the flow body so reviewers can see checkpoints like:

- `sandboxed_pydantic_ai_agent_model_request`
- `run_sandbox_command_tool`
- `sandboxed_pydantic_ai_agent_model_request_2`
- `publish_sandbox_answer`

Because those adapter-created model/tool checkpoints are terminal graph steps in this demo shape, the script does **not** call `.wait()` to extract a final Python value. Instead it submits with `cache=False`, polls `handle.status` until the execution finishes, prints the execution ID, and tells the reviewer where to inspect the named checkpoints.

### Reproduction

Focused automated path:

```bash
uv run pytest tests/test_pydantic_ai_sandbox_toolset.py tests/test_pydantic_ai_adapter.py tests/test_sandbox_runtime.py
just example-coverage-audit
just check
just test
```

Manual example path, once the base sandbox-stack branch is usable in your environment:

```bash
uv sync --extra local --extra pydantic-ai --extra openai
uv run kitaru init
export OPENAI_API_KEY=sk-...
uv run python examples/integrations/pydantic_ai_agent/pydantic_ai_sandbox_toolset.py
```

The example builds a `KitaruAgent` with `sandbox_command_toolset(...)` and asks the model to run `python --version` through the active stack sandbox. You should see an execution ID plus instructions to inspect the checkpoints in the Kitaru UI or with `kitaru executions get <exec_id>`. The final text is stored on the `publish_sandbox_answer` checkpoint.

### Local checks run

```text
uv run pytest tests/test_pydantic_ai_sandbox_toolset.py
# 23 passed, 1 warning

just example-coverage-audit
# Example coverage audit passed.

just check
# All checks passed.

just test
# 2544 passed, 13 skipped, 86 warnings
```

Oracle review and simplify review both completed. The final Oracle pass after simplify found no blockers.

## Dependency note

This PR targets `feature/sandbox-stack-component-base` and depends on the temporary ZenML sandbox-enabled dependency gate already documented there. Before the combined sandbox work is merged/released, the ZenML git SHA should still be replaced with the released `zenml==0.95.0` dependency as described in the base PR.

### CODE REVIEW

File: examples/integrations/pydantic_ai_agent/pydantic_ai_sandbox_toolset.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+stderr, and exit code, then cleans up the session.
+
+Prerequisites:
+    uv sync --extra local --extra pydantic-ai --extra openai
+    uv run kitaru init
+    export OPENAI_API_KEY=sk-...
+
+Your active stack must have exactly one sandbox component. The default local
+Kitaru stack includes one when the installed ZenML version supports sandbox
+components.
```
[the remark this answers is not in the document]
Comment (strickvl) [reply examples/integrations/pydantic_ai_agent/pydantic_ai_sandbox_toolset.py]: Addressed in 9861cd11 by rephrasing the setup note in Kitaru terms: the example now refers to the installed Kitaru local runtime supporting sandbox components, without mentioning ZenML.
